DOCS-1195: Review modular resources (#1992)
viamrobotics/docs: docs/appendix/glossary/modular-resource.md

---
title: Modular Resource
id: modular-resource

> full_link: https://docs.viam.com/modular-resources/key-concepts/#resources

Reviewer: You use this glossary entry most of the time to replace the link to the top level page, so that should be what we link to here
```suggestion
full_link: https://docs.viam.com/modular-resources/
```

short_description: A modular resource is a custom component, service, or logic implemented through Viam's public APIs.
aka:
---

> A modular resource is a custom {{< glossary_tooltip term_id="component" text="component" >}}, {{< glossary_tooltip term_id="service" text="service" >}}, or logic implemented through [Viam's public APIs](/extend/modular-resources/key-concepts/#valid-apis-to-implement-in-your-model).

Reviewer: We do have another definition here which could also use a bit of work but I think the following suggestion works:  

```suggestion
A modular resource is a custom {{< glossary_tooltip term_id="model" text="model" >}} of an existing {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} that implements the same [API methods](https://docs.viam.com/program/apis/).
```

> A modular resource is a custom {{< glossary_tooltip term_id="model" text="model" >}} of an existing {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} that implements the same [API methods](https://docs.viam.com/program/apis/).

Reviewer: bit worried that the existing might confuse people
```suggestion
A modular resource is a custom {{< glossary_tooltip term_id="model" text="model" >}} of a {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} that implements the same [API methods](https://docs.viam.com/program/apis/).
```

> For more information see the [Modular Resource Documentation](/extend/modular-resources/).

Reviewer: this resource has moved
```suggestion
For more information see the [Modular Resource Documentation](/modular-resources/).
```
